src/hotspot/share/classfile/classLoaderExt.cpp

```diff
@@ -100,4 +100,8 @@ void ClassLoaderExt::record_result(const s2 classpath_index, InstanceKlass* resu
     CDSConfig::disable_heap_dumping();
   }
 #endif // INCLUDE_CDS_JAVA_HEAP
+
+  if (CDSConfig::is_dumping_preimage_static_archive() || CDSConfig::is_dumping_dynamic_archive()) {
+    AOTClassLocationConfig::dumptime()->check_invalid_classpath_index(classpath_index, result);
```

Comment:
In case the `classpath_index` is invalid, I don't think we should call `AOTClassLocationConfig::dumptime_update_max_used_index()`.  Maybe the `check_invalid_classpath_index()` function should return a bool and have `ClassLoaderExt::record_result()` update the classpath index and max used index.

Reply:
Fixed.